src/lib/core/mp_compression.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "mp_compression.h"
+
+#include <stddef.h>
+
+#include "trivia/config.h"
+#include "trivia/util.h"
+
+#if defined(ENABLE_TUPLE_COMPRESSION)
+# error unimplemented
+#endif
+
+char *
+mp_compress(char *dst, const char *src, size_t src_size,
+	    enum compression_type type)
+{
+	(void)dst;
+	(void)src;
+	(void)src_size;
+	(void)type;
+	unreachable();
+	return NULL;
+}
+
+size_t
+mp_decompress(const char **src, char *dst, size_t dst_size)
+{
+	(void)src;
+	(void)dst;
+	(void)dst_size;
+	/** Zero is an error according to the convention. */
+	return 0;
+}
```

Comment:
Fixed, but as see in all others stubs commits we implemet *.c files not static inline functions in header files.

Reply:
Not all others. I usually follow the following pattern:
1. If a stub depends on something (e.g. diag or say), I put in in a *.c file to avoid `#include` in the header. This isn't strictly necessary, but I wanted to give an example how to implement stubs in *.c files:
   https://github.com/tarantool/tarantool/blob/c7823c5d1e1a5f9108285106c5dd680c5e3cf514/src/lib/core/ssl.c#L18-L25
2. If a stub is unreachable no-op, I put it in the header file. I hope that this way a coverage scan will see that this code isn't supposed to be executed and won't include it in coverage (I'm not sure it works though):
   https://github.com/tarantool/tarantool/blob/c7823c5d1e1a5f9108285106c5dd680c5e3cf514/src/lib/core/ssl.h#L27-L32